### DISCUSSION TITLE
Prevent background update for data when network is offline.

### DISCUSSION THREAD

**Discussion by bonatdev** [vercel/swr#647] [General]
I am using PWA with CRA, 
so is it possible if the network was offline to prevent replacing old data with the new on, in this case, error, because the network is offline?

**Comment by pke** [comment]
I think you can easily implement that with a check in you fetch function. 

```js
function networkAwareFetch(staleData, ...args) {
  return isOnline()
    .then(() => fetch(...args))
    .catch(() => staleData)
}
```

**Reply by sergiodxa** [reply]
Instead of passing staleData to the fetcher you could get it from the cache with
```js
import { cache } from "swr"
function networkAwareFetch(...key) {
  return isOnline()
    .then(() => fetch(...args))
    .catch(() => cache.get(key))
}
```

**Reply by pke** [reply]
Cool. I didn't know about the `cache`. Is this a new export? 
Also the fetcher is now bound to swr, while my solution could be plugged into any framework. Also your function signature seems to be off, where is `...args` coming from now?

**Reply by sergiodxa** [reply]
Cache has been exported for a while, but it's still not documented, has been pretty stable tho.

About the `...args` is because I copied it from your example and forgot to edit it, that should be `...key`.

And yes, it will bound it to SWR.

**Comment by JaneJeon** [comment]
I myself create a middleware that returns stale `data` and empty `error` if we detect a network error.
